api/src/convert/collateral.rs

```diff
@@ -24,12 +24,25 @@ impl TryFrom<&Collateral> for external::Collateral {
 impl TryFrom<&external::Collateral> for Collateral {
     type Error = ConversionError;
     fn try_from(src: &external::Collateral) -> Result<Self, Self::Error> {
-        let bytes = encode_to_protobuf_vec(src)?;
-        let prost = prost::Collateral::decode(bytes.as_slice())?;
+        let prost = prost::Collateral::from(src);
         Ok((&prost).try_into()?)
     }
 }
 
+impl From<&external::Collateral> for prost::Collateral {
+    fn from(src: &external::Collateral) -> Self {
+        Self {
```

Comment:
I think using the `Self{}` here will ensure we don't omit a field and prost only has the fields unlike protobufs so this shouldn't be too bad to maintain.

Implementation wise these are both structures of nested byte vectors, or strings, so there really shouldn't be any issue conerting between the two, it's when one wants the `mc-sgx-dcap-types::Collateral` that it can fail.